pkg/snapshot/restorer/restorer_test.go

```diff
@@ -215,17 +629,24 @@ func checkDataConsistency(dir string, logger *logrus.Logger) error {
 	for currKey := keyFrom; currKey <= keyTo; currKey++ {
 		key = keyPrefix + strconv.Itoa(currKey)
 		value = valuePrefix + strconv.Itoa(currKey)
-
 		resp, err := cli.Get(context.TODO(), key, opts...)
 		if err != nil {
 			return fmt.Errorf("unable to get value from etcd: %v", err)
 		}
 		if len(resp.Kvs) == 0 {
-			return fmt.Errorf("entry not found for key %s", resKey)
+			//handles deleted keys as every 10th key is deleted during populate etcd call
+			if math.Mod(float64(currKey), 10) == 0 {
+				continue //it should continue as key was put for action delete
+			} else {
+				res := resp.Kvs[0]
+				resKey = string(res.Key)
+				return fmt.Errorf("entry not found for key %s", resKey)
```

Comment:
```suggestion
				return fmt.Errorf("entry not found for key %s", key)
```
I agree there was a bug previously where we used to pass `resKey` here instead of `key`.